From 888ca33b571d99e877d665235b822f7c961c8fdb Mon Sep 17 00:00:00 2001
From: "R. Tyler Croy" <tyler@monkeypox.org>
Date: Thu, 28 Aug 2014 16:24:04 -0700
Subject: [PATCH 6/8] Update some headers to include the right headers to build
 on FreeBSD

---
 src/rd.h     | 9 +++++++++
 src/rdaddr.h | 4 ++++
 2 files changed, 13 insertions(+)

diff --git a/src/rd.h b/src/rd.h
index c31501e..4789493 100644
--- a/src/rd.h
+++ b/src/rd.h
@@ -37,7 +37,11 @@
 #include <errno.h>
 #include <time.h>
 #include <sys/time.h>
+
+#ifndef __FreeBSD__
+/* alloca(3) is in stdlib on FreeBSD */
 #include <alloca.h>
+#endif
 #include <assert.h>
 #include <pthread.h>
 
@@ -110,6 +114,11 @@
 # endif
 #endif /* sun */
 
+#ifdef __FreeBSD__
+/* FreeBSD defines be64toh() in sys/endian.h */
+#include <sys/endian.h>
+#endif
+
 #ifndef be64toh
 #ifndef __APPLE__
 #ifndef sun
diff --git a/src/rdaddr.h b/src/rdaddr.h
index 0b37354..e55bd55 100644
--- a/src/rdaddr.h
+++ b/src/rdaddr.h
@@ -32,6 +32,10 @@
 #include <arpa/inet.h>
 #include <netdb.h>
 
+#ifdef __FreeBSD__
+#include <sys/socket.h>
+#endif
+
 /**
  * rd_sockaddr_inx_t is a union for either ipv4 or ipv6 sockaddrs.
  * It provides conveniant abstraction of AF_INET* agnostic operations.
-- 
1.9.0

